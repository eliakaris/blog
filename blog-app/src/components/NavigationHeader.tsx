import * as React from 'react';
import { Link } from 'react-router-dom';

function NavigationHeader() {
  return (
  <div>
    <div className="App-header">
        <h2>Welcome to React!!</h2>
    </div>
    <nav>
      <ul>
        <li><Link to="/blog/first-blog-item">First blog item</Link></li>
        <li><Link to="/blog/second-blog-item">Second blog item</Link></li>
      </ul>
    </nav>
  </div>);
}

export default NavigationHeader;