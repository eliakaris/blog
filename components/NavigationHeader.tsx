import React, { useState } from 'react';
import NavLink from 'next/link';
import {
  Collapse,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler
} from 'reactstrap';

function NavigationHeader() {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <nav className="navbar navbar-expand-lg navbar-light fixed-top" id="mainNav">
      <div className="container">
        <NavbarBrand href="/">Elia Karagiannis</NavbarBrand>
        <NavbarToggler onClick={() => setIsOpen(!isOpen)} />
        <Collapse navbar isOpen={isOpen}>
          <Nav className="ml-auto" navbar>
            <NavItem><NavLink href="/">Home</NavLink></NavItem>
            <NavItem><NavLink href="/about">About</NavLink></NavItem>
            <NavItem><NavLink href="/blog">Archives</NavLink></NavItem>
          </Nav>
        </Collapse>
      </div>
    </nav>
  );
};

export default NavigationHeader;
