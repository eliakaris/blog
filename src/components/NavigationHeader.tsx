import * as React from 'react';
import { NavLink } from 'react-router-dom';
import {
  Collapse,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler } from 'reactstrap';

interface State {
  isOpen: boolean;
}

interface Props {
}

class NavigationHeader extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { isOpen: false };
  }

  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  public render() {
    return (
      <nav className="navbar navbar-expand-lg navbar-light fixed-top" id="mainNav">
        <div className="container">
          <NavbarBrand href="/">Elia Karagiannis</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse navbar isOpen={this.state.isOpen}>
            <Nav className="ml-auto" navbar>
              <NavItem><NavLink to="/">Home</NavLink></NavItem>
              <NavItem><NavLink to="/about">About</NavLink></NavItem>
              <NavItem><NavLink to="/blog">Archive</NavLink></NavItem>
            </Nav>
          </Collapse>
        </div>
      </nav>);
  }
}

export default NavigationHeader;