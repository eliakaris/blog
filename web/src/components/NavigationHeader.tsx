import React, { useState } from 'react';
import { NavLink } from '../packages/react-router-dom';
import {
  Collapse,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler } from '../packages/reactstrap';

function NavigationHeader() {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <nav className="navbar navbar-expand-lg navbar-light fixed-top" id="mainNav">
      <div className="container">
        <NavbarBrand href="/">Elia Karagiannis</NavbarBrand>
          <NavbarToggler onClick={() => setIsOpen(!isOpen)} />
          <Collapse navbar isOpen={isOpen}>
            <Nav className="ml-auto" navbar>
              <NavItem><NavLink to="/">Home</NavLink></NavItem>
              <NavItem><NavLink to="/about">About</NavLink></NavItem>
              <NavItem><NavLink to="/blog">Archive</NavLink></NavItem>
            </Nav>
          </Collapse>
      </div>
    </nav>
  );
};

export default NavigationHeader;
