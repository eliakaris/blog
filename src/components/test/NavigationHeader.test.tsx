import * as React from 'react';
import NavigationHeader from '../NavigationHeader';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

configure({ adapter: new ReactSixteenAdapter() });

describe('<NavigationHeader /> tests', () => {
  let renderer: any;

  beforeAll(() => {
    renderer = ShallowRenderer.createRenderer();
  });

  it('renders a snapshot', () => {
    const component = renderer.render(<NavigationHeader />);
    expect(component).toMatchSnapshot();
  });
});