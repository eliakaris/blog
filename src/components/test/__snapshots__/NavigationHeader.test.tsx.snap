// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationHeader /> tests renders a snapshot 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
  >
    <NavbarHeader>
      <NavbarBrand>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
        >
          Elia Karagiannis
        </NavLink>
      </NavbarBrand>
      <NavbarToggle />
    </NavbarHeader>
    <NavbarCollapse>
      <Nav
        bsClass="nav"
        justified={false}
        pullLeft={false}
        pullRight={true}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
          >
            Home
          </NavLink>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/about"
          >
            About
          </NavLink>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/blog"
          >
            Archive
          </NavLink>
        </NavItem>
      </Nav>
    </NavbarCollapse>
  </Uncontrolled(Navbar)>
</div>
`;
